receiver/dockerstatsreceiver/factory.go

```diff
@@ -22,15 +22,25 @@ import (
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.opentelemetry.io/collector/service/featuregate"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver/internal/metadata"
 )
 
 const (
-	typeStr   = "docker_stats"
-	stability = component.StabilityLevelAlpha
+	typeStr              = "docker_stats"
+	stability            = component.StabilityLevelAlpha
+	newImplFeatureGateID = "receiver.dockerstats.useNewImplementation"
```

Comment:
Your java is bleeding through here :P, `impl` isn't something that is encouraged in go.

I would suggest naming it `featureGateDockerStats` just so it is clear on what it is.

Reply:
`featureGateDockerStats` doesn't explain what it is, it isn't specific enough to what the actual featuregate does. Is "implementation" really a bad word to use? There's two implementations of `scrape()` and this featuregate switches between them. Maybe it would be better to name it `useNewScraper`?

Reply:
`useScraperV2` maybe?

Reply:
`useScraperV2` is good, I've changed it to that 👍